lib/plug/static.ex

```diff
@@ -179,10 +184,11 @@ defmodule Plug.Static do
 
   defp serve_static({:ok, conn, file_info, path}, segments, options) do
     %{qs_cache: qs_cache, et_cache: et_cache, et_generation: et_generation,
-      gzip?: gzip?, brotli?: brotli?, headers: headers} = options
+      gzip?: gzip?, brotli?: brotli?, headers: headers, mime_types: types} = options
     case put_cache_header(conn, qs_cache, et_cache, et_generation, file_info, path) do
       {:stale, conn} ->
-        content_type = segments |> List.last |> MIME.from_path
+        filename = List.last(segments)
+        content_type = types[filename] || MIME.from_path(filename)
```

Comment:
I think it'd be better to keep `||` operator, I believe it will be faster, and every bit counts for static serving. :bowtie: